eng/common/TestResources/New-TestResources.ps1

```diff
@@ -426,6 +382,82 @@ try {
+            }
+
+            $global:AzureTestPrincipal = $servicePrincipal
+            $global:AzureTestSubscription = $SubscriptionId
+
+            Log "Created service principal '$($AzureTestPrincipal.ApplicationId)'"
+            $AzureTestPrincipal
+            $resourceGroupRoleAssigned = $true
+        }
+
+        $TestApplicationId = $servicePrincipal.ApplicationId
+        $TestApplicationOid = $servicePrincipal.Id
+        $TestApplicationSecret = (ConvertFrom-SecureString $servicePrincipal.Secret -AsPlainText)
+    }
+
+    # Get test application OID from ID if not already provided. This may fail if the
+    # provisioner is a service principal without permissions to query AAD. This is a
+    # critical failure, but we should prompt with possible remediation.
+    if ($TestApplicationId -and !$TestApplicationOid) {
+        Log "Attempting to query the Object ID for the test service principal"
+
+        try {
+            $testServicePrincipal = Retry {
+                Get-AzADServicePrincipal -ApplicationId $TestApplicationId
+            }
+        }
+        catch {
+            Write-Warning "The Object ID of the test application was unable to be queried. You may want to consider passing it explicitly with the 'TestApplicationOid` parameter."
+            throw $_.Exception
+        }
+
+        if ($testServicePrincipal -and $testServicePrincipal.Id) {
+            $script:TestApplicationOid = $testServicePrincipal.Id
+        }
+    }
+
+    # Make sure pre- and post-scripts are passed formerly required arguments.
+    $PSBoundParameters['TestApplicationId'] = $TestApplicationId
+    $PSBoundParameters['TestApplicationOid'] = $TestApplicationOid
+    $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
+
+    # Grant the test service principal ownership over the resource group. This may fail if the provisioner is a
+    # service principal without permissions to grant RBAC roles to other service principals. That should not be
+    # considered a critical failure, as the test application may have subscription-level permissions and not require
+    # the explicit grant.
+    if (!$resourceGroupRoleAssigned) {
+        Log "Attempting to assigning the 'Owner' role for '$ResourceGroupName' to the Test Application '$TestApplicationId'"
+        $principalOwnerAssignment = New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId "$TestApplicationId" -ResourceGroupName "$ResourceGroupName" -ErrorAction SilentlyContinue
+
+        if ($principalOwnerAssignment.RoleDefinitionName -eq 'Owner') {
+            Write-Verbose "Successfully assigned ownership of '$ResourceGroupName' to the Test Application '$TestApplicationId'"
+        } else {
+            Write-Warning "The 'Owner' role for '$ResourceGroupName' could not be assigned. You may need to manually grant the role to the Test Application '$TestApplicationId' if it does not have subscription-level permissions."
```

Comment:
Which role are we saying must be manually granted here?

Reply:
The `Owner` role;   I was assuming the context from the beginning of the warning would have been clear.    I reworded to be more explicit.   Does that read better?